galley/pkg/runtime/projections/serviceentry/convert/convert.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package convert
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	networking "istio.io/api/networking/v1alpha3"
+	"istio.io/istio/galley/pkg/runtime/log"
+	"istio.io/istio/galley/pkg/runtime/projections/serviceentry/annotations"
+	"istio.io/istio/galley/pkg/runtime/projections/serviceentry/node"
+	"istio.io/istio/galley/pkg/runtime/projections/serviceentry/pod"
+	"istio.io/istio/galley/pkg/runtime/resource"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/serviceregistry/kube"
+
+	coreV1 "k8s.io/api/core/v1"
+)
+
+// Annotations augments the annotations for the k8s Service.
+func Annotations(serviceAnnotations resource.Annotations) resource.Annotations {
+	out := make(resource.Annotations)
+	for k, v := range serviceAnnotations {
+		out[k] = v
+	}
+
+	out[annotations.SyntheticResource] = "true"
+	return out
+}
+
+// Service converts a k8s Service to a networking.ServiceEntry
+func Service(spec *coreV1.ServiceSpec, metadata resource.Metadata, fullName resource.FullName, domainSuffix string) *networking.ServiceEntry {
+	resolution := networking.ServiceEntry_STATIC
+	location := networking.ServiceEntry_MESH_INTERNAL
+	endpoints := convertExternalServiceEndpoints(spec, metadata)
+
+	// Check for an external service
+	externalName := ""
+	if spec.Type == coreV1.ServiceTypeExternalName && spec.ExternalName != "" {
+		externalName = spec.ExternalName
+		resolution = networking.ServiceEntry_DNS
+		location = networking.ServiceEntry_MESH_EXTERNAL
+	}
+
+	// Check for unspecified Cluster IP
+	addr := model.UnspecifiedIP
+	if spec.ClusterIP != "" && spec.ClusterIP != coreV1.ClusterIPNone {
+		addr = spec.ClusterIP
+	}
+	if addr == model.UnspecifiedIP && externalName == "" {
+		// Headless services should not be load balanced
+		resolution = networking.ServiceEntry_NONE
+	}
+
+	ports := make([]*networking.Port, 0, len(spec.Ports))
+	for _, port := range spec.Ports {
+		ports = append(ports, convertPort(port))
+	}
+
+	namespace, name := convertFullName(fullName)
+	se := &networking.ServiceEntry{
+		Hosts:      []string{serviceHostname(name, namespace, domainSuffix)},
+		Addresses:  []string{addr},
+		Resolution: resolution,
+		Location:   location,
+		Ports:      ports,
+		Endpoints:  endpoints,
+		ExportTo:   convertExportTo(metadata.Annotations),
```

Comment:
where is the subject alt name? https://github.com/istio/api/blob/release-1.1/networking/v1alpha3/service_entry.proto#L497

Reply:
The only way that I'm aware of to get subject alt names for a service is via annotations.

In previous discussions with @costinm it sounded like those annotations weren't really used anymore.